keylime/payloads/keylime-controller/local_action_node_failover.py

```diff
@@ -1,27 +1,23 @@
 import os
 import asyncio
+import subprocess
 
-async def execute(revocation):
-    """Node fail over script
+"""Node fail over script
 
-    This script will be sent to the keylime_agent running on a machine
-    which has an OpenShift client (oc) installed and configured.
+This script will be sent to the keylime_agent running on a machine
+which has an OpenShift client (oc) installed and configured.
 
-    When a node is compromised this script will run `oc adm cordon` and
-    `oc adm drain` commands to remove that worker node from the OpenShift
-    cluster
-    """
-    if revocation['type']!='revocation':
-        return
+When a node is compromised this script will run `oc adm cordon` and
+`oc adm drain` commands to remove that worker node from the OpenShift
+cluster
+"""
 
-    # TODO - get the node name
-    node_name = "XXX"
-    rc = os.system("oc adm cordon {}".format(node_name))
-    if rc != 0:
-        print("Could not cordon node {}: Command failed with code {}".format(node_name, rc))
+async def execute(revocation):
+    if revocation['type']!='revocation':
         return
 
-    rc = os.system("oc adm drain {}".format(node_name))
-    if rc != 0:
-        print("Could not drain node {}: Command failed with code {}".format(node_name, rc))
-        return
+    node_name = 'mpeters-okd-zxrh5-' + revocation['agent_id']
+    cmd = "KUBECONFIG=/etc/kubeconfig /usr/local/bin/oc adm cordon {}".format(node_name)
+    subprocess.call(cmd, shell=True)
```

Comment:
Should we use `run` here instead and capture stdout/stderr and log to a file somewhere? I understand `call` is part of the older API but can still be used for compatibility reasons.

Reply:
Yeah, I went back and forth on that. Bug logging from within a revocation action didn't seem straightforward to me, not without changing the keylime code to pass the logger into the `execute()` command. It would be nice to have more control over that, but if the execute fails it will bubble up the chain and be logged in at least a basic way. There are definitely ways to improve the dev experience of writing local_actions for keylime and making logging easier would be one of them.